app/controllers/thredded/application_controller.rb

```diff
@@ -88,10 +88,17 @@ def pundit_user
       thredded_current_user
     end
 
+    # Returns the `@messageboard` instance variable.
+    # If `@messageboard` is not set, it first sets it to the topic with the slug or ID given by
+    # `params[:messageboard_id]`.
+    #
+    # @return [Thredded::Messageboard]
+    # @raise [Thredded::Errors::MessageboardNotFound] if the topic with the given slug does not exist.
     def messageboard
-      @messageboard ||= params[:messageboard_id].presence && Messageboard.friendly.find(params[:messageboard_id])
-    rescue ActiveRecord::RecordNotFound
-      raise Thredded::Errors::MessageboardNotFound
+      @messageboard ||= begin
+        fail Thredded::Errors::MessageboardNotFound unless params[:messageboard_id].presence
```

Comment:
Bit of a nit pick, but I feel a bit uncomfortable with this style (hard to read) (previous version was odd too IMO). 

Also -- not sure why, but the behaviour seems to change in this PR. Previously it was **not** raising MessageboardNotFound when messageboard_id empty, now it is...

Instead maybe extract the check to a `required_messageboard_id` or `messageboard_id!` (?!) method  which does `params[:messageboard_id].presence || fail Thredded::Errors::MessageboardNotFound` Will make this method super simple